2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,210 @@
+
+```python
+lass Trainee(AbstractBaseUser):
+    '''Model for a Trainee '''
+    # listing just essential fields, more fields can and should be added when implementing
+    COURSE_CHOICES = (
+        ('1', 'SWC'),
+        ('2', 'DC'),
+        )
+    personal    = models.CharField(max_length=STR_LONG)
+    middle      = models.CharField(max_length=STR_LONG, null=True, blank=True)
+    family      = models.CharField(max_length=STR_LONG)
+    email       = models.CharField(max_length=STR_LONG, unique=True, null=True, blank=True)
+    course      = models.CharField(max_length=STR_LONG, choices=COURSE_CHOICES, null=False, default='1')
+    group       = models.ForeignKey(Trainee_Group,default=None,null=True)
+    Attendance_Flag = models.BooleanField(default=False) # maintains attendance record
+    Discussion_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+    HomeWork_Flag = models.BooleanField(default=False) # maintains PASS or FAIL
+
+    username    = models.CharField(
+        max_length=STR_MED, unique=True,
+        validators=[RegexValidator(r'^[\w\-_]+$', flags=re.A)],
+    )
+ ```
+
+Maintaining Group information:
+
+```python
+class Trainee_Group(models.Model):
+
+    instructor_assigned=models.ForeignKey(Person)
+    primary_contact_personal    = models.CharField(max_length=STR_LONG)
+    primary_contact_middle      = models.CharField(max_length=STR_LONG, null=True, blank=True)
+    primary_contact_family      = models.CharField(max_length=STR_LONG)
+    primary_contact_email       = models.CharField(max_length=STR_LONG, unique=True, null=True, blank=True)
+    training_course             = models.CharField(max_length=STR_LONG)
+```
+Commenting System:
+Since comments/feedback are only meant to be unidirectional (i.e from the discussion_leaders, lesson_maintainers to the trainee) a simple Commenting Model would suffice and there is no need for third party apps.
+
+```python
+
+class D_Comments(models.Model):
+    ''' commenting system
+    storing comments by the discussion leaders
+    '''
+    Discussion_Comment =models.TextField(default="", blank=True)
+    Discussion_leader = models.ForeignKey(Person,default=None,null=True)
+    trainee = models.ForeignKey(Trainee,default=None,null=True) # Many to one relationship
+    created = models.DateTimeField(auto_now_add=True)
+
+class H_Comments(models.Model):
+    ''' commenting System
+    storing comments by the lesson maintainers
+    '''
+    Homework_Comment = models.TextField(default="", blank=True)
+    lesson_maintainer = models.ForeignKey(Person,default=None,null=True)
+    trainee =  models.ForeignKey(Trainee,default=None,null=True) #Many to one relationship
+    created = models.DateTimeField(auto_now_add=True)
+
+```
+
+
+## Schedule of Deliverables
+
+I believe in test drive development so I will be writing tests as soon as I am done with a feature/view addition.
```

Comment:
"test driven" - can you please provide a pointer to a publicly-visible repository containing code you have developed using TDD?